src/cider/nrepl/middleware/track_state.clj

```diff
@@ -281,21 +345,32 @@
          all-namespaces (if cljs
                           (vals (cljs-ana/all-ns cljs))
                           (all-ns))
-         project-ns-map (fast-reduce (fn [acc ns]
-                                       (let [name (ns-name-fn ns)]
-                                         (if (jar-ns-fn name)
-                                           acc ;; Remove all jar namespaces.
-                                           (assoc! acc name (ns-as-map ns all-namespaces)))))
-                                     all-namespaces)
+         ns-as-map-fn (memoize ns-as-map)
```

Comment:
By memoizing ns-as-map which ultimately calls the new, IO-costly stuff, we reuse that IO work a few times.

The memoization is not global - its lifespan is that of the nrepl request.